src/core/utils/fs.cpp

```diff
@@ -439,18 +439,30 @@ QString Utils::Fs::QDesktopServicesCacheLocation()
 
 QString Utils::Fs::QDesktopServicesDownloadLocation()
 {
-#if defined(Q_OS_WIN) || defined(Q_OS_OS2)
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+#if defined(Q_OS_WIN)
+    if (QSysInfo::windowsVersion() <= QSysInfo::WV_XP)  // Windows XP
+        return QDir(QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation)).absoluteFilePath(
+                    QCoreApplication::translate("fsutils", "Downloads"));
+#endif
+    return QStandardPaths::writableLocation(QStandardPaths::DownloadLocation);
+#else
+
+#if defined(Q_OS_OS2)
     // as long as it stays WinXP like we do the same on OS/2
     // TODO: Use IKnownFolderManager to get path of FOLDERID_Downloads
     // instead of hardcoding "Downloads"
     // Unfortunately, this would break compatibility with WinXP
```

Comment:
No don't delete. Transfer it below to WINDOWS case. Or make this WINDOWS case and below OS/2 case.